src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -6569,26 +6570,41 @@ address generate_avx_ghash_processBlocks() {
       const Register j = r9;
       const Register k = r10;
       const Register l = r11;
+      const Register table = r14;
 #ifdef _WIN64
       const Register y = rdi;
       const Register z = rsi;
 #else
       const Register y = rcx;
       const Register z = r8;
 #endif
-      assert_different_registers(crc, buf, len, a, j, k, l, y, z);
+      assert_different_registers(crc, buf, len, a, j, k, l, y, z, table);
 
       BLOCK_COMMENT("Entry:");
       __ enter(); // required for proper stackwalking of RuntimeStub frame
 #ifdef _WIN64
       __ push(y);
       __ push(z);
 #endif
```

Comment:
a, j,  y and z are only required on the crc32c_ipl_alg2_alt2() path, so should be initialized and saved/restored only there.  
This will also help you to pick a  save on call register like c_rarg3 for table without having to push/pop.

Reply:
Done.  Changed to use `j` instead of save/restore of `r14`.